python-lib/wordcloud_generator.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+"""Module with a class to generate wordclouds based on cleaned text"""
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+import matplotlib
+import matplotlib.pyplot as plt
+
+matplotlib.use("agg")
+
+import traceback
+from typing import List, AnyStr
+import pandas as pd
+import spacy.lang
+import os
+from wordcloud import WordCloud
+from collections import Counter
+from time import time
+from spacy_tokenizer import MultilingualTokenizer
+from plugin_io_utils import count_records
+import random
+
+
+class WordcloudGenerator:
+    """Class to generate multilingual wordclouds based on text data and save them as png images
+    Attributes:
+        df (pandas.DataFrame): Dataframe containing text data
+        tokenizer (MultilingualTokenizer): Tokenizer used for text processing
+        text_column (str): Name of the df column on which to compute wordclouds
+        language (str, optional): Language in which to tokenize data (for monolingual wordcloud), defaults to english.
+            Use "Detected language column" for multilingual tokenization
+        language_column (str, optional): Name of the language column
+        subchart_column (str, optional): Name of the subcharts column to compute wordclouds on, defaults to None
+        max_words (int, optional): Maximum number of words to display in wordcloud, defaults to 100
+    """
+
+    DEFAULT_MAX_WORDS = 100
+    DEFAULT_NUM_PROCESS = 2
```

Comment:
This is not used, right?

Reply:
max words is used line 82, num process is indeed unnecessary